tracks/julia/exercises/pythagorean-triplet/mentoring.md

```diff
@@ -25,34 +55,51 @@ function pythagorean_triplets(n)
     return triplets
 end
 ```
-````
 
-Telling Julia the type of `triplets` allows it to store the values without indirection. Without an annotation some (all?) version of Julia will type `triplets` as `Vector{Any}`, which means a vector of pointers to heap-allocated objects, which is slower.
+Calculating c from a and b clearly eliminates a huge number of iterations and is the main reason for the speedup.
 
-A small improvement is to tighten up the loop bounds using some simple arithmetic and ceiling division (`cld`):
+A much more modest improvement is achieved by telling Julia the type of `triplets` so that it can store the values without indirection.
+Without an annotation some (all?) version of Julia will type `triplets` as `Vector{Any}`, which means a vector of pointers to heap-allocated objects, which is slower.
+Annotating with an immutable type like we have done (which by definition will be of known size) allows Julia to simply store the items one after the other without indirection.
```

Comment:
Another speed-up can be achieved by using `sizehint!`. However, I'm not sure if there's a formula to estimate the size in advance without overshooting too much. Do you know a way?